### PR TITLE
[fix](Nereids): log constraint editlog in lock scope

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run builldall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import org.apache.doris.common.DdlException;
 import org.apache.doris.common.MetaNotFoundException;
 import org.apache.doris.nereids.exceptions.AnalysisException;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java:28]: maybe we should not throw nereids exception in catalog class

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java (hunk 1 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java:28]: Yeah, we can log error rather than throw exception.

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**